src/core/src/op/variadic_split.cpp

```diff
@@ -33,81 +34,15 @@ void ngraph::op::v1::VariadicSplit::validate_and_infer_types() {
-
-            auto split_lengths = split_lengths_constant->cast_vector<int64_t>();
-            // Adjust split lengths in case of negatives
-            int64_t sum_of_splits = 0;
-            int64_t negative_one = -1;
-            for (size_t i = 0; i < split_lengths.size(); i++) {
-                NODE_VALIDATION_CHECK(this,
-                                      split_lengths[i] >= -1,
-                                      "Invalid value ",
-                                      split_lengths[i],
-                                      " in split lengths input. Should be >= -1.");
-
-                if (split_lengths[i] == -1) {
-                    NODE_VALIDATION_CHECK(this,
-                                          negative_one == -1,
-                                          "Cannot infer split with multiple -1 values at ",
-                                          negative_one,
-                                          " and ",
-                                          i);
-                    negative_one = i;
-                } else {
-                    sum_of_splits += split_lengths[i];
-                }
-            }
-            const auto data_shape_dims = vector<Dimension>{data.get_partial_shape()};
-            const auto dimension_at_axis = data_shape_dims.at(axis);
-
-            if (negative_one >= 0 && dimension_at_axis.is_static()) {
-                split_lengths[negative_one] = dimension_at_axis.get_length() - sum_of_splits;
-                sum_of_splits += split_lengths[negative_one];
-            }
-            if (data_shape[axis].is_static()) {
-                NODE_VALIDATION_CHECK(this,
-                                      sum_of_splits == data_shape[axis].get_length(),
-                                      "Total length of splits: ",
-                                      sum_of_splits,
-                                      " must match the length of the chosen axis: ",
-                                      data_shape[axis]);
-            }
-
-            for (int64_t output{0}; output < num_outputs; ++output) {
-                const auto output_split_dim =
-                    split_lengths.at(output) == -1 ? Dimension::dynamic() : split_lengths.at(output);
-                auto tmp_shape = data_shape_dims;
-                tmp_shape.at(axis) = output_split_dim;
-                set_output_type(output, data_type, ov::PartialShape{tmp_shape});
-            }
-        } else {
-            for (int64_t output{0}; output < num_outputs; ++output) {
-                set_output_type(output, data_type, ov::PartialShape::dynamic());
-            }
-        }
+    std::vector<ov::PartialShape> input_shapes = {get_input_partial_shape(0),
+                                                  get_input_partial_shape(1),
+                                                  get_input_partial_shape(2)};
+    std::vector<ov::PartialShape> output_shapes;
+    shape_infer(this, input_shapes, output_shapes, {});
+
```

Comment:
In the current implementation we won't infer `output_shapes` in case when `split_lengths_pshape` is dynamic because we don't know about numbers of outputs. But in the `validate_and_infer_types()` we can know number of outputs via `get_output_size()`.
Could you please modify the `validate_and_infer_types()` to handle this corner case (And leaving the `shape_infer` function unchanged)?

Reply:
validate_and_infer_types() is supposed to call set_output_size() since it also responsible for infer number of output, you search `set_output_size` in `./src/core/src/op` folder, many ops call this function inside their `validate_and_infer_types()`.